### ISSUE TITLE
Extend broadcast functionalities

### ISSUE THREAD

**Issue by srdante** [ddnet/ddnet#4897]
The idea is to extend the broadcast functionality by making the following detailed changes:

## Handle correctly the break lines on broadcast (Client):

- Currently the broadcast is centered on screen. However, if you send a broadcast with `\n`, the aligment will respect the size of the last line, when all lines should be centered by every line size independently (As happens on Teeworlds 0.7).
- At the moment the ddnet client does not automatically break the broadcast into multiple lines if it doesn't fit on the screen. The break lines should happen so that this problem does not occur (As happens on Teeworlds 0.7).

**Details to discuss:**

- How will it be calculated when the client will have to break line automatically (% margin from right screen border?; only on screen border?;will it break by letter or word?).

## Add native bottom broadcast (Client & Server):

- The net message would be `CNetMsg_Sv_Broadcast_Bottom` with the same structure as the default one. 
- This can open up the possibility  to be able to show 2 broadcasts at the same time (Default broadcast on top, and some player special statuses for mods on bottom).
- The server side will need to check if `Client Version` newer or equal `Version that implemented the bottom broadcast`, otherwise will send the default broadcast message even for bottom _**(or an adapted one with a lot of \n)**_.
- If the client is 0.7 both broadcasts will show on bottom (As it currently happens).

**Details to discuss:**

- Will the broadcast be shown glued to the bottom like 0.7 or should it be shown at the same margin distance from bottom as the common broadcast?

## Add colors support for broadcast (Client & Server):

- Implement the functionality to colorize specific parts of the broadcast will be implemented in both the standard broadcast and the bottom broadcast.
- The implementation is intended to be similar to teeworlds 0.7, where it is a native implementation.
- The server side will need to sanitize the broadcast message to remove the colors flags if  `Client Version`  is older than `Version that implemented colored broadcast`.
- If the client is 0.7 there will be no problem, as this is a native feature there.

**Details to discuss:**

- At the moment I don't have any details to discuss, since we have a model that we can follow which is the 0.7 version.
- However, I tested the color broadcast using the current ddnet server version, and in 0.7 it does not show the characters referring to the colors (As it shows on 0.6) but the text remains white. Futher investigation is needed.

**How the broadcast is colorized in Teeworlds 0.7:**
```cpp
CNetMsg_Sv_Broadcast chatMsg;
char msgBuff[256];
str_format(msgBuff, sizeof(msgBuff), "WAVE ^980%d ^999(%d zombies)", waveID + 1,
		m_WaveSpawnCount[waveID]);
chatMsg.m_pMessage = msgBuff;
Server()->SendPackMsg(&chatMsg, MSGFLAG_VITAL, -1);
```

**Screenshot with the result on Zomb 0.7 version (From Teeworlds forum, didn't tested by myself):**

![image](https://user-images.githubusercontent.com/35383529/160224675-545b133c-6f2f-469e-bd88-9b0213b7ba39.png)

**Comment by def-** [comment]
I‘m fine with enabling 0.7 style broadcasts, can probably also take some of the code.

Is this something you‘re interested in for your mods? DDNet itself doesn‘t use any fancy broadcasts, just map announcements

**Comment by srdante** [comment]
@def- Yes. These broadcast changes is mainly focused for mods. Colored broadcast could also fits for normal broadcast by servers owners and DDNet.

An hypothetical example for a use of DDNet would be to announce an event with the date highlighted in green.
